src/gennylib/src/context.cpp

```diff
@@ -130,6 +139,10 @@ std::unordered_map<PhaseNumber, std::unique_ptr<PhaseContext>> ActorContext::con
 }
 
 mongocxx::pool::entry ActorContext::client() {
+    if (_hasApmOpts) {
+        // TODO: Remove this conditional when CDRIVER-2931 is resolved.
+        return _workload->_clientPool->acquire();
+    }
```

Comment:
I think you can move this method to `WorkloadContext` and just delegate to it. This would prevent you from having to pass down the boolean.


```c++
mongocxx::pool::entry ActorContext::client() {
  return _workload->client();
}
```